DownloadManager/Plugin.DownloadManager.Android/DownloadFileImplementation.cs

```diff
@@ -92,20 +94,44 @@ public DownloadFileImplementation (ICursor cursor)
             Url = cursor.GetString (cursor.GetColumnIndex (Android.App.DownloadManager.ColumnUri));
         }
 
-        public void StartDownload (Android.App.DownloadManager downloadManager, string destinationPathName)
+        private bool _mobileNetworkAllowed = true;
+
+        public bool MobileNetworkAllowed
         {
-            using (var downloadUrl = Uri.Parse (Url))
-            using (var request = new Android.App.DownloadManager.Request (downloadUrl)) {
+            get
+            {
+                return _mobileNetworkAllowed;
+            }
+            set
+            {
+                _mobileNetworkAllowed = value;
+                if (Request != null && value == false) {
+                    Request.SetAllowedNetworkTypes(Android.App.DownloadNetwork.Wifi);
```

Comment:
The target framework should be on one of the latest because you can use it downwards. In the app project the developer should set a minimum target to ensure it is available. I would advise setting it to API lvl 24.